### PR TITLE
Manager changes for isolation for tutorials/courses

### PR DESCRIPTION
**Pull request by sagark**
This adds a tutorial mode for the manager (will also be useful for classes) that externally sets a bunch of parameters based on tags applied to the manager instance. This is merely for configuration of the manager -- enforcement of these things happens in IAM policies.

This sets a per-user:
vpcname
securitygroup
keyname
snsname
s3bucketname

I have not tested this heavily, this PR is just for early feedback right now.

### CODE REVIEW

**Comment by sagark** [comment]
@davidbiancolin this is ready for review/merge

File: deploy/awstools/awstools.py [hunk 1]
```diff
@@ -74,6 +142,13 @@ def launch_instances(instancetype, count, instancemarket, spotinterruptionbehavi
 
          This will launch instances in avail zone 0, then once capacity runs out, zone 1, then zone 2, etc.
     """
+
+    # now, check if we're in tutorial mode and override if necessary
```
Comment (davidbiancolin) [inline deploy/awstools/awstools.py]: This comment no longer applies right?

File: deploy/runtools/runtime_config.py [hunk 2]
```diff
@@ -244,8 +244,18 @@ def __init__(self, runtimeconfigfile, configoverridedata):
...
             runtime_dict[overridesection][overridefield] = overridevalue
 
-        runfarmtagprefix = "" if 'FIRESIM_RUNFARM_PREFIX' not in os.environ else os.environ['FIRESIM_RUNFARM_PREFIX'] + "-"
+        runfarmtagprefix = "" if 'FIRESIM_RUNFARM_PREFIX' not in os.environ else os.environ['FIRESIM_RUNFARM_PREFIX']
+        if runfarmtagprefix != "":
+            runfarmtagprefix += "-"
+
         self.runfarmtag = runfarmtagprefix + runtime_dict['runfarm']['runfarmtag']
+
+        # now, check if we're in tutorial mode and set extra prefix if so
```
Comment (davidbiancolin) [inline deploy/runtools/runtime_config.py]: Ditto

File: deploy/runtools/runtime_config.py [hunk 3]
```diff
@@ -244,8 +244,18 @@ def __init__(self, runtimeconfigfile, configoverridedata):
             rootLogger.warning(overridefield + "=" + overridevalue)
             runtime_dict[overridesection][overridefield] = overridevalue
 
-        runfarmtagprefix = "" if 'FIRESIM_RUNFARM_PREFIX' not in os.environ else os.environ['FIRESIM_RUNFARM_PREFIX'] + "-"
+        runfarmtagprefix = "" if 'FIRESIM_RUNFARM_PREFIX' not in os.environ else os.environ['FIRESIM_RUNFARM_PREFIX']
```
Comment (davidbiancolin) [inline deploy/runtools/runtime_config.py]: Remind me, how is this being set in the environment?

Comment (sagark) [reply deploy/runtools/runtime_config.py]: This is set by the user if they want, we don't set it anywhere explicitly. It's just to help users keep track if they have multiple managers/clones of firesim.

Comment (sagark) [reply deploy/runtools/runtime_config.py]: (this feature was added before, I just fixed a bug where if you don't have this env var set, your runfarmtag would start with a "-")

**Pull request merged**